esp/logging/logginglib/logthread.hpp

```diff
@@ -41,7 +41,16 @@ class CLogRequestReaderSettings : public CSimpleInterface
 
 class CLogThread;
 
-class CLogRequestReader : public CInterface, implements IThreaded
+interface ILogRequestReader : extends IThreaded
+{
+    virtual CLogRequestReaderSettings* getSettings() = 0;
+    virtual void setPause(bool pause) = 0;
+    virtual void reportAckedLogFiles(StringArray& ackedLogFiles) = 0;
+    virtual void removeUnknownAckedLogFiles(StringArray& ackedLogFiles) = 0;
+    virtual void cleanAckedLogFiles(StringArray* fileNames) = 0;
```

Comment:
minor: filenames cannot be null, so it should really be a reference parameter

Reply:
Ok, I will change it to reference parameter.